store/models/service_agreement.go

```diff
@@ -165,22 +169,32 @@ type Encumbrance struct {
 func (e Encumbrance) ABI() ([]byte, error) {
 	buffer := bytes.Buffer{}
 	var paymentHash common.Hash
+	var err error
+	write := func(buf []byte) {
+		if err != nil {
+			return
+		}
+		_, err = buffer.Write(buf)
+	}
 	if e.Payment != nil {
 		paymentHash = e.Payment.ToHash()
 	}
-	_, err := buffer.Write(paymentHash.Bytes())
-	if err != nil {
-		return []byte{}, err
-	}
+	write(paymentHash.Bytes())
```

Comment:
Can we avoid the anonymous `func` and just do it inline?
```
_, err := buffer.Write(paymentHash.Bytes())
if err != nil { return err }
```

We're also not returning `err` if an error does happen. We seem to silently fail `write` and continue on with the method.

Reply:
Thanks for all the feedback. I'm agnostic about `write`. I only did it that way was because [Code Climate complained about having a function with more than four `return`s](https://github.com/smartcontractkit/chainlink/pull/672/commits/106f580cf50d49a9b5c65e1ba949c035e3b7653f). I noticed Code Climate has a "won't fix" option. Should I back this change out ignore Code Climate?

It's surprising that Code Climate has fussy rules like this, but nothing about `XXX` comments (though I should have caught that myself; thanks for catching it for me.)

Reply:
Yes ignore Code Climate for the complain about more than four returns. It's more important that we handle the error, if any, and returning errors is idiomatic go. Perhaps we could make a whole new method but it's not worth it in this scenario.

Reply:
OK, I removed 106f580c.

Reply:
Do I need extra permissions to mark the Code Climate issue "wontfix"? I don't see the drop-down menu described in the [docs](https://docs.codeclimate.com/docs/issues).

Reply:
You can just ignore it for now. It doesn't block merging.